### PR TITLE
Allow running test by workflow name

### PR DESCRIPTION
**Pull request by rmarrcode**
I tested the changes on a few different scenarios. It wouldn't hurt to make sure that these changes were effective.

### CODE REVIEW

File: touca/_workflow.py [hunk 1]
```diff
@@ -164,7 +170,10 @@ def _update_testcase_list(options: dict):
         return
     if "testcase_file" in options:
         with open(options["testcase_file"], "rt") as file:
-            keep = lambda x: x and not x.startswith("#")
+
+            def keep(x):
+                return x and not x.startswith("#")
+
```
Comment (ghorbanzade) [inline touca/_workflow.py:176]: Any reason why we should use a nested function instead of a lambda?

File: touca/_workflow.py [hunk 2]
```diff
@@ -58,6 +58,10 @@ def _parse_cli_options(args) -> Dict[str, Any]:
         help="Version of the code under test")
     parser.add_argument("--suite", metavar='',
         help="Slug of suite to which test results belong")
+    parser.add_argument("--workflow", metavar='',
+        help='Used to explicity define workflow as suite'
```
Comment (ghorbanzade) [inline touca/_workflow.py:62]: What do you think of "Name of the workflow to run" as an alternative for the help message?

**Review by ghorbanzade — COMMENTED** [review]
LGTM. Made minor comments to get your thoughts.

File: touca/_workflow.py (hunk 2 above)
Comment (rmarrcode) [reply touca/_workflow.py:62]: This is a good idea. I think I wrote this before I really understood what the ticket was about.

**Pull request merged**